detection_rules/packaging.py

```diff
@@ -250,6 +253,38 @@ def save(self, verbose=True):
         if verbose:
             click.echo('Package saved to: {}'.format(save_dir))
 
+    def export(self, outfile, downgrade_version=None, verbose=True, skip_unsupported=False):
+        """Export rules into a consolidated ndjson file."""
+        outfile = Path(outfile).with_suffix('.ndjson')
+        unsupported = []
+
+        if downgrade_version:
+            if skip_unsupported:
+                export_str = ''
+
+                for rule in self.rules:
+                    try:
+                        export_str += json.dumps(downgrade_contents_from_rule(rule, downgrade_version),
```

Comment:
i think this logic across the branches would be simpler if there was a single call to `outfile.write_text("\n".join(output_lines)` at the end, and you just buffer these to an array